src/components/VmStatus/getVmStatus.js

```diff
@@ -156,17 +131,138 @@ const isWaitingForVmi = vm => {
+      diskAction = 'Uploading';
+      break;
+    case DATA_VOLUME_SOURCE_URL:
+      diskAction = 'Importing';
+      break;
+    default:
+      break;
+  }
+  return diskAction ? `${diskAction} disk ${diskName} failed` : `Disk ${diskName} failed`;
+};
+
+export const isPreparingDisks = (vm, cdiPods, dataVolumes) => {
+  const diskStatuses = [];
+  if (!dataVolumes) {
+    return NOT_HANDLED;
+  }
+  const dataVolumeVolumes = getVolumes(vm).filter(v => v.dataVolume);
+  dataVolumeVolumes.forEach(volume => {
+    const dVolume = dataVolumes.find(
+      dv => getName(dv) === volume.dataVolume.name && getNamespace(dv) === getNamespace(vm)
+    );
+    if (!dVolume) {
+      return;
+    }
+    const pod = findCdiPod(volume, cdiPods);
+    const podError = getPodError(pod);
+    const diskStatus = {
+      diskName: volume.name,
+      pod,
+    };
+    if (podError) {
+      diskStatuses.push({
+        ...diskStatus,
+        status: VM_STATUS_DISKS_FAILED,
+        message: getDiskErrorMessage(dVolume, volume.name),
+        verboseMessage: podError,
+      });
+    } else {
+      switch (get(dVolume, 'status.phase')) {
+        case DATA_VOLUME_STATUS_PENDING:
+        case DATA_VOLUME_STATUS_PVC_BOUND:
+          diskStatuses.push({
+            ...diskStatus,
+            status: VM_STATUS_PREPARING_DISKS,
+            diskStatus: DATA_VOLUME_STATUS_PENDING,
+          });
+          break;
+        case DATA_VOLUME_STATUS_CLONE_SCHEDULED:
+        case DATA_VOLUME_STATUS_CLONE_IN_PROGRESS:
+          diskStatuses.push({
+            ...diskStatus,
+            status: VM_STATUS_PREPARING_DISKS,
+            diskStatus: DATA_VOLUME_STATUS_CLONE_IN_PROGRESS,
+          });
+          break;
+        case DATA_VOLUME_STATUS_UPLOAD_SCHEDULED:
+        case DATA_VOLUME_STATUS_UPLOAD_IN_PROGRESS:
+          diskStatuses.push({
+            ...diskStatus,
+            status: VM_STATUS_PREPARING_DISKS,
+            diskStatus: DATA_VOLUME_STATUS_UPLOAD_IN_PROGRESS,
+          });
+          break;
+        case DATA_VOLUME_STATUS_IMPORT_SCHEDULED:
+        case DATA_VOLUME_STATUS_IMPORT_IN_PROGRESS:
+          diskStatuses.push({
+            ...diskStatus,
+            status: VM_STATUS_PREPARING_DISKS,
+            diskStatus: DATA_VOLUME_STATUS_IMPORT_IN_PROGRESS,
+          });
+          break;
+        case DATA_VOLUME_STATUS_FAILED:
+          diskStatuses.push({
+            ...diskStatus,
+            status: VM_STATUS_DISKS_FAILED,
+            diskStatus: DATA_VOLUME_STATUS_FAILED,
+            message: getDiskErrorMessage(dVolume, volume.name),
+            verboseMessage: `Failed preparing ${volume.name} disk`,
+          });
+          break;
+        default:
+          break;
+      }
+    }
+  });
+  if (diskStatuses.length === 1) {
+    return diskStatuses[0];
+  }
+  if (diskStatuses.length > 1) {
+    const failedDiskStatuses = diskStatuses.filter(ds => ds.status === VM_STATUS_DISKS_FAILED);
```

Comment:
shouldn't DATA_VOLUME_STATUS_FAILED statuses be also included?

Reply:
DATA_VOLUME_STATUS_FAILED is still picked up because im checking status field VM_STATUS_DISKS_FAILED which indicates that 'something' failed and that 'something' can be either data volume ( DATA_VOLUME_STATUS_FAILED ) or pod